pkgs/by-name/s7/s7/package.nix

```diff
@@ -0,0 +1,159 @@
+{
+  lib,
+  fetchFromGitLab,
+  stdenv,
+  notcurses,
+  gmp,
+  mpfr,
+  libmpc,
+  flint3,
+  unstableGitUpdater,
+  writeScript,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "s7";
+  version = "11.2-unstable-2024-11-02";
+
+  src = fetchFromGitLab {
+    domain = "cm-gitlab.stanford.edu";
+    owner = "bil";
+    repo = "s7";
+    rev = "6b6651de318c3360aa2711a445659f372ffe894e";
+    hash = "sha256-9x1u60jflESAHgP0gwIDVFwhI5XU5gsAiXaic8zdW2U=";
+  };
+
+  buildInputs = [
+    notcurses
+    gmp
+    mpfr
+    libmpc
+    flint3
+  ];
+
+  # The following scripts are modified from [Guix's](https://packages.guix.gnu.org/packages/s7/).
+
+  patchPhase = ''
+    runHook prePatch
+
+    substituteInPlace s7.c \
+        --replace-fail libc_s7.so $out/lib/libc_s7.so
+
+    runHook postPatch
+  '';
```

Comment:
Please use `prePatch` or `postPatch` instead.  Specifying a custom `patchPhase` almost always ignores patches specified by `patches = [ ... ];`.

https://github.com/NixOS/nixpkgs/blob/70cfc062e07f7da2256fb8d04905a45041387366/pkgs/stdenv/generic/setup.sh#L1327-L1352

Reply:
I see. I'd forgotten that `patchPhase` is more likely to be overridden than `buildPhase` and `installPhase`, whose controlling attributes only suit for `make` based scripts.